### ISSUE TITLE
[MongoDB] Introspection does not work with MongoDB Atlas Serverless clusters: `Error in connector: Error querying the database: An invalid argument was provided: loadBalanced is not supported`

### ISSUE THREAD

**Issue by janpio** [prisma/prisma#9680]
When trying to run `prisma db pull` on a connection string for a MongoDB Atlas Serverless cluster/database, it does not work yet:

```
λ npx prisma db pull
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db"

Introspecting based on datasource defined in prisma\schema.prisma …

Error: Error in connector: Error querying the database: An invalid argument was provided: loadBalanced is not supported
```

Reproduction is easy:
1. Create MongoDB Atlas Serverless cluster at https://cloud.mongodb.com/
2. Create new folder and `cd` into it
3. `npm init -y`
4. `npm install prisma @prisma/client`
5. `npx prisma init --datasource-provider mongodb`
6. Add connection string from Atlas to `.env` file
7. `npx prisma db pull`
8. Observe error message above

Tested with Prisma CLI 3.2.1

---

The error message `loadBalanced is not supported` points to this code in the MongoDB Rust driver we are using:

```
        // Disallow use of load-balanced configurations in non-test code.
        // TODO RUST-653 Remove this when load balancer work is ready for release.
        if !cfg!(test) && self.load_balanced.is_some() {
            return Err(ErrorKind::InvalidArgument {
                message: "loadBalanced is not supported".to_string(),
            }
            .into());
        }
```

https://github.com/mongodb/mongo-rust-driver/blob/5a0192cf1ba11d9e44c453e14f90d32231689166/src/client/options/mod.rs#L1152-L1159

Issue referenced there is https://jira.mongodb.org/projects/RUST/issues/RUST-653?filter=allopenissues

So seems Atlas Serverless (with MongoDB 5) is using `loadBalanced` somehow that their driver does not support yet.

**Comment by janpio** [comment]
There is an update on the JIRA issue: https://jira.mongodb.org/browse/RUST-653?focusedCommentId=4132461&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4132461
Seems this was added to the driver: https://github.com/mongodb/mongo-rust-driver/pull/495
No release of the driver since then though.

**Comment by dpetrick** [comment]
Fixed on latest dev (~ `3.5.0-dev.14`). Will be released with 3.5.0.
